### PR TITLE
[3party] Optionally allow building with packaged protobuf

### PR DESCRIPTION
**Pull request by Ferenc-**
Signed-off-by: Ferenc Géczi <ferenc.gm@gmail.com>

### CODE REVIEW

**Comment by biodranik** [comment]
Is protobuf updated to the latest version? Are there any issues with the latest version?

**Comment by Ferenc-** [comment]
> Is protobuf updated to the latest version? Are there any issues with the latest version?

Now, that the macro has been dropped, the version has no impact on this PR, so those questions can be the topic of the separate update PR.

File: indexer/CMakeLists.txt [hunk 1]
```diff
@@ -149,8 +149,6 @@ file(COPY ${OTHER_FILES} DESTINATION ${CMAKE_BINARY_DIR})
 
 omim_add_library(${PROJECT_NAME} ${SRC})
 
-add_subdirectory(${OMIM_ROOT}/3party/protobuf ${CMAKE_CURRENT_BINARY_DIR}/protobuf)
```
Comment (biodranik) [inline indexer/CMakeLists.txt:152]: Isn't the second param necessary?

**Review by biodranik — COMMENTED** [review]
Now we can expect reports about "it doesn't build if I use system package".

**Comment by Ferenc-** [comment]
> Now we can expect reports about "it doesn't build if I use system package".

We can set up a CI with a reference build setup with system packages, and just point towards it.
If a user can't replicate that for whatever reason, then it is not our problem anymore.

File: indexer/CMakeLists.txt (hunk 1 above)
Comment (Ferenc-) [reply indexer/CMakeLists.txt:152]: > Isn't the second param necessary?

Necessary for what?
For the build to work? Apparently not.
For some so far (for me) unclear purpose, that you might know about?
Let me know and I put it back immediately.

Comment (biodranik) [reply indexer/CMakeLists.txt:152]: Is it related to generated protoc binary that can be used to recompile changed map styles?

Comment (Ferenc-) [reply indexer/CMakeLists.txt:152]: > generated protoc binary 

The `protoc` binary does not appear to be generated by the current `cmake` + `ninja` build process, on the `master` branch:
````bash
find -iname \*proto\*
./indexer/protobuf
./indexer/protobuf/CMakeFiles/protobuf.dir
./OMaps.app/Contents/Resources/drules_proto_dark.bin
./OMaps.app/Contents/Resources/drules_proto_vehicle_clear.bin
./OMaps.app/Contents/Resources/drules_proto_clear.bin
./OMaps.app/Contents/Resources/drules_proto_vehicle_dark.bin
./libprotobuf.a
./drules_struct.proto
````
Have I missed something perhaps?



> recompile changed map styles

From the `STYLES.md` I don't see how and where that is needed, from that it appears that only the already generated python output is used. Once protobuf is istalled with `pip install protobuf`.

The one place where I have found usage of `protoc` is the `generate_proto.sh`, but that just assumes, that the `protoc` is somewhere in the `PATH`.

Comment (biodranik) [reply indexer/CMakeLists.txt:152]: As far as I remember, protoc shoud be the same version to generate styles properly. @vng @pastk do you install some custom protoc/protobuf version or the latest one works fine with styles generation?

Comment (pastk) [reply indexer/CMakeLists.txt:152]: > As far as I remember, protoc shoud be the same version to generate styles properly. @vng @pastk do you install some custom protoc/protobuf version or the latest one works fine with styles generation?

I don't seem to have a `protoc` binary in my system even. For the styles compiler to work I just installed python protobuf as per the STYLES doc. ATM I'm using `3.19.4`.

Comment (Ferenc-) [reply indexer/CMakeLists.txt:152]: How about you @vng ?

Comment (vng) [reply indexer/CMakeLists.txt:152]: I work on MacOS. brew install protobuf is enough for me.

**Review by biodranik — COMMENTED** [review]
Please rebase and check if it works with the latest system protobuf library.

**Review by biodranik — APPROVED** [review]

**Comment by Ferenc-** [comment]
@biodranik Thanks for the approval, I don't have merge permissions here, so let me know if there is anything I can do.

**Pull request merged**